Coolbooru/Coolbooru.cs

```diff
@@ -157,63 +378,155 @@ public class CoolImageQuery {
+		/// </summary>
+		public bool IncludeFavoritedBy = false;
+		/// <summary>
+		/// When specified, sorts the images randomly.
+		/// </summary>
+		public bool SortRandomly;
 	}
 
 	/// <summary>
 	/// Represents arguments to a user gallery query.
 	/// </summary>
 	public class CoolUserGalleryQuery {
-		public string user;
-		public int? id;
-		public string key;
-		public int page = 1;
-		public bool include_images = false;
+		/// <summary>
+		/// Your Derpibooru API key.
+		/// </summary>
+		public string APIKey;
+		/// <summary>
+		/// THe username of the user to pull galleries from.
+		/// </summary>
+		public string User;
+		/// <summary>
+		/// The ID of the gallery to pull. Optional.
+		/// </summary>
+		public int? ID;
+		/// <summary>
+		/// The page of results to return. Defaults to 1.
+		/// </summary>
+		public int Page = 1;
+		/// <summary>
+		/// Whether or not to return an array of image IDs featured in each gallery, in the order defined by the owner. Disregards content filters.
+		/// </summary>
+		public bool IncludeImages = false;
 	}
 
 	/// <summary>
-	/// Represents an OEmbed response.
+	/// Represents an oEmbed response.
 	/// </summary>
 	public class CoolEmbed {
+		/// <summary>
+		/// The version of oEmbed used.
+		/// </summary>
 		public string version;
+		/// <summary>
+		/// The type of oEmbed response.
+		/// </summary>
 		public string type;
+		/// <summary>
+		/// The title parameter of the oEmbed structure.
+		/// </summary>
 		public string title;
+		/// <summary>
+		/// The author's URL.
+		/// </summary>
 		public string author_url;
+		/// <summary>
+		/// The author's name (specified by artist: tag).
+		/// </summary>
 		public string author_name;
+		/// <summary>
+		/// The provider name (usually Derpibooru).
+		/// </summary>
 		public string provider_name;
+		/// <summary>
+		/// The URL of the image page on Derpibooru.
+		/// </summary>
 		public string provider_url;
+		/// <summary>
+		/// A Unix timestamp that specifies an age for caching purposes. Derpibooru recommends you cache your oEmbed responses!
+		/// </summary>
 		public int cache_age;
+		/// <summary>
+		/// The Derpibooru ID of the image.
+		/// </summary>
 		public int derpibooru_id;
+		/// <summary>
+		/// The net score on Derpibooru for the image.
+		/// </summary>
 		public int derpibooru_score;
+		/// <summary>
+		/// The number of comments on the image.
+		/// </summary>
 		public int derpibooru_comments;
+		/// <summary>
+		/// A list of tags applied to the image.
+		/// </summary>
 		public List<string> derpibooru_tags;
+		/// <summary>
+		/// The direct link to the image.
+		/// </summary>
 		public string thumbnail_url;
 	}
 
 	/// <summary>
 	/// The main functions of Coolbooru.
 	/// </summary>
-	public class Coolbooru {
-
+	public class CoolRequests {
```

Comment:
I don't necessarily agree with this title, it makes it sound like it's a representation of something. Maybe CoolMethods would be a better name? (this is the main thing blocking a merge)